src/btree/node.rs

```diff
@@ -260,30 +250,35 @@ impl Node {
 		if at {
 			let existing = self.separator_address(i);
 			if let Some(existing) = existing {
-				Column::write_existing_value_plan::<_, Vec<u8>>(
+				if Column::write_existing_value_plan::<_, Vec<u8>>(
 					&TableKey::NoHash,
 					values,
 					existing,
 					change,
 					log,
 					None,
-				)?;
-			}
-			let _ = self.remove_separator(i);
-			if depth != 0 {
-				// replace by bigger value in left child.
-				if let Some(mut child) = self.fetch_child(i, values, log)? {
-					let (need_balance, sep) = child.remove_last(depth - 1, values, log)?;
-					self.write_child(i, child, values, log)?;
-					if let Some(sep) = sep {
-						self.set_separator(i, sep);
-					}
-					if need_balance {
-						self.rebalance(depth, i, values, log)?;
+					false,
+				)?
+				.0
+				.is_none()
```

Comment:
return type of this function is pretty terrible (multiple meaning depending on the way the function is called), yet I did not change it in this pr.